stdlib/REPL/docs/src/index.md

```diff
@@ -52,6 +52,32 @@ If it is enabled, you can try it out by pasting the code block above this paragr
 the REPL. This feature does not work on the standard Windows command prompt due to its limitation
 at detecting when a paste occurs.
 
+Objects are printed at the REPL using the [`show`](@ref) function with a specific [`IOContext`](@ref).
+In particular, the `:limit` attribute is set to `true`.
+Other attributes can receive in certain `show` methods a default value if it's not already set,
+like `:compact`.
+It's possible to specify the attributes used by the REPL via the `Base.active_repl.iocontext`
```

Comment:
Is this the first time `Base.active_repl` is introduced? In that case, it might make sense to create a dedicated function for this in the `REPL` module.

Reply:
It seems to be the first time that `Base.active_repl` is introduced. I'm not clear of the point of a dedicated function to access it, and I think it's fine allow it for now (this field is likely to stay for a while). That said, I should probably document how to do that in the startup file.

Reply:
Well, the point is to not expose some weird global variable in Base that is only defined in at a certain point in the startup procedure.

Reply:
Also, there is an `options` field in `active_repl`  (unfortunately not yet documented), whose fields can be interactively explored with tab completion. AFAICT, if `active_repl` was accessible only via a call to a function, there would not be autocompletion.

Reply:
If we put this in the documentation then the `active_repl.iocontext` is an official API and it can never ever be changed. It seems likely we want to have some better way to set `options` in the REPL than this in the future so committing to this "setting fields in a global variable in Base API" doesn't seem that great.

If there isn't a good way of setting options in the REPL right now I would prefer merging without exposing the current implementation as a stable API.

Reply:
> the point is to not expose some weird global variable in Base

I would normally agree, but this is an interactive thing, so it doesn't seem as necessary. If there was a function, I would just never use it, and access directly `active_repl`. And having a function won't help with the fact that it's only defined at a certain point in the startup procedure...
If you insist, I will remove the reference to `active_repl` (or say that it's an internal/experimental feature) and just leave the freshly added doc which explains how to use `iocontext` from the startup file. But I won't define the function `get_active_repl() = Base.active_repl`.

Reply:
> But I won't define the function get_active_repl() = Base.active_repl.

Heh, no.

My thinking is something along the lines of having:

```
REPL.options.iocontext = Dict(:compat => false)
REPL.options.bell = false
...
```

where `REPL.options` is documented (and tab completable).

The active repl creates a reference to the `options` object when it gets initialized and the `options` object is always defined.

Reply:
OK, this is beyond the scope of this PR though. Do you suggest that `REPL.options` be global? It could be an object with overloaded `setproperty!` and `getproperty`, to make it a proxy to `Base.active_repl.options`... Indeed looks like it may be better than `Base.active_repl`.

> If we put this in the documentation then the active_repl.iocontext is an official API and it can never ever be changed.

OK, this was not my understanding, as it's interactive. So what about, in this PR, leaving the `atreplinit` portion of the doc as the official way, and pointing at the `Base.active_repl` as an experimental way to play with the options?

Reply:
Sure, if we denote it as experimental, that's fine.

Reply:
So I just marked the whole feature as experimental anyway, as once it's used, we may find out a better way to achieve the same, but we just need to start somewhere.

Reply:
@KristofferC Just FYI `Module.CONFIG` is now used in at least three packages for this kind of purpose (disclaimer: I did the implementation in all cases):

https://github.com/queryverse/ElectronDisplay.jl
https://github.com/JuliaDebug/Cthulhu.jl
https://github.com/tkf/InteractiveCodeSearch.jl

Although I guess I'll follow stdlib approach if `REPL.options` is implemented.